projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php

```diff
@@ -222,7 +222,20 @@ public function add_tools_menu( $wp_admin_import = false, $wp_admin_export = fal
 	 * @param bool $wp_admin Optional. Whether links should point to Calypso or wp-admin. Default false (Calypso).
 	 */
 	public function add_options_menu( $wp_admin = false ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		add_menu_page( esc_attr__( 'Settings', 'jetpack' ), __( 'Settings', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/general/' . $this->domain, null, 'dashicons-admin-settings', 80 );
+		$slug = 'https://wordpress.com/settings/general/' . $this->domain;
+		add_menu_page( esc_attr__( 'Settings', 'jetpack' ), __( 'Settings', 'jetpack' ), 'manage_options', $slug, null, 'dashicons-admin-settings', 80 );
+		add_submenu_page( $slug, esc_attr__( 'General', 'jetpack' ), __( 'General', 'jetpack' ), 'manage_options', $slug );
+		add_submenu_page( $slug, esc_attr__( 'Security', 'jetpack' ), __( 'Security', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/security/' . $this->domain );
+		add_submenu_page( $slug, esc_attr__( 'Performance', 'jetpack' ), __( 'Performance', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/performance/' . $this->domain );
+		add_submenu_page( $slug, esc_attr__( 'Writing', 'jetpack' ), __( 'Writing', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/writing/' . $this->domain );
+		add_submenu_page( $slug, esc_attr__( 'Discussion', 'jetpack' ), __( 'Discussion', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/discussion/' . $this->domain );
+
+		$has_scan     = \Jetpack_Plan::supports( 'scan' );
+		$rewind_state = get_transient( 'jetpack_rewind_state' );
+		$has_backup   = $rewind_state && in_array( $rewind_state->state, array( 'awaiting_credentials', 'provisioning', 'active' ), true );
+		if ( $has_scan || $has_backup ) {
+			add_submenu_page( $slug, esc_attr__( 'Jetpack', 'jetpack' ), __( 'Jetpack', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/jetpack/' . $this->domain );
+		}
```

Comment:
This is based on how Calypso currently decides whether the Jetpack section should be rendered: https://github.com/Automattic/wp-calypso/blob/19b9e60a68f6bcc821b669f6e9051e51ef4607fc/client/my-sites/site-settings/navigation.jsx#L117-L122

Reply:
@mmtr Do I understand correctly that there should be a "Jetpack" submenu option under "Settings" if that JP connected site has Backup/Scan features on it?

![Screen Shot 2021-06-18 at 3 24 58 PM](https://user-images.githubusercontent.com/15803018/122618476-91130700-d04b-11eb-9885-6c14dcd21dca.png)

If so, I'm not seeing one there on a JN site I'm testing with. The site has a JP Complete plan, and is actively scanning files and backing up. Logging out `$has_scan` or `$has_backup` are both empty for me.

Reply:
Good catch. Yeah, you should see the Jetpack submenu in that scenario. I'll see what's going on here

Reply:
@samiff You're probably dealing with a caching issue. If you go first to `/wp-admin/admin.php?page=jetpack#/settings` and then reload Calypso, you should see the Jetpack submenu.

The Rewind state is cached on a transient and only updated when explicitly forced with a `_cacheBuster` param: https://github.com/Automattic/jetpack/blob/281e2544bff1a305f4809ee763a269c8069448e0/projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php#L1587-L1612

Due to performance concerns, I think it's safer to keep rely on the cached state, otherwise we may be fetching from WP.com the current Rewind state on every Calypso request.

Reply:
After visiting `/wp-admin/admin.php?page=jetpack#/settings` I was able to see the Jetpack submenu.